exporters/trace/util/src/main/java/io/opencensus/exporter/trace/util/TimeLimitedHandler.java

```diff
@@ -0,0 +1,120 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.trace.util;
+
+import com.google.common.util.concurrent.SimpleTimeLimiter;
+import com.google.common.util.concurrent.TimeLimiter;
+import com.google.errorprone.annotations.MustBeClosed;
+import io.opencensus.common.Duration;
+import io.opencensus.common.Scope;
+import io.opencensus.trace.Sampler;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Status;
+import io.opencensus.trace.TraceOptions;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.export.SpanData;
+import io.opencensus.trace.export.SpanExporter;
+import io.opencensus.trace.export.SpanExporter.Handler;
+import io.opencensus.trace.samplers.Samplers;
+import java.util.Collection;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * An abstract class that allows different tracing services to export recorded data for sampled
+ * spans in their own format within a given time frame. If export does not complete within the time
+ * framework, spans will be dropped.
+ *
+ * <p>Only extend this class if the client APIs don't support timeout natively. If there is a
+ * timeout option in the client APIs (for example Stackdriver Trace V2 API allows you to set
+ * timeout), use that instead.
+ *
+ * <p>To export data this MUST be register to to the ExportComponent using {@link
+ * SpanExporter#registerHandler(String, Handler)}.
+ *
+ * @since 0.22
+ */
+public abstract class TimeLimitedHandler extends SpanExporter.Handler {
+
+  private static final Logger logger = Logger.getLogger(TimeLimitedHandler.class.getName());
+  private static final Tracer tracer = Tracing.getTracer();
+  private static final Sampler lowProbabilitySampler = Samplers.probabilitySampler(0.0001);
+
+  private final Duration deadline;
+  private final String exportSpanName;
+
+  protected TimeLimitedHandler(Duration deadline, String exportSpanName) {
+    this.deadline = deadline;
+    this.exportSpanName = exportSpanName;
+  }
+
+  /**
+   * Exports a list of sampled (see {@link TraceOptions#isSampled()}) {@link Span}s using the
+   * immutable representation {@link SpanData}, within the given {@code deadline} of this {@link
+   * TimeLimitedHandler}.
+   *
+   * @param spanDataList a list of {@code SpanData} objects to be exported.
+   * @since 0.22
+   */
+  public abstract void timeLimitedExport(Collection<SpanData> spanDataList);
+
+  @Override
+  public void export(final Collection<SpanData> spanDataList) {
+    final Scope exportScope = newExportScope();
+    try {
```

Comment:
nit: how about change this to

```java
try (exportScope = tracer.spanBuilder(exportSpanName)
           .setSampler(lowProbabilitySampler)
           .startScopedSpan()) {
   ...
} catch (...) {
   ...
}
```
and remove
```
 @MustBeClosed
  private Scope newExportScope() {
    return tracer.spanBuilder(exportSpanName).setSampler(lowProbabilitySampler).startScopedSpan();
  }
```

Reply:
Compatibility of this package is JDK 1.6: https://github.com/census-instrumentation/opencensus-java/pull/1887/files#diff-86539a8dfa2c3e85249a9a31286e2673R4.

This is required by a few exporters, e.g:
https://github.com/census-instrumentation/opencensus-java/blob/f68f73b5926b97ada0c6952f31856e2a9e0e0de0/exporters/trace/jaeger/build.gradle#L4

Reply:
got it.

Reply:
I asked exactly the same thing before :)